docs/guide/modularize.md

```diff
@@ -0,0 +1,44 @@
+# Modularize your build function
+
+Envd allows you to easily create reusable components, so that the building file can be more organized.
+
+## Defining your own package
+
+envd support most Python syntax, including string format, for-loop, if-statement, function definition. 
+
+For example, you can define a function for tensorflow package like this
+```python
+def tensorfloow(version):
+    if version.startswith("1"):
+        # execute command to install tf1.x
+        run(["pip install tensorflow-gpu=={}".format(version)])
+    else:
+        # execute command to install tf2.x
+        run(["pip install tensorflow=={}".format(version)]) 
+```
+The full list of language spec can be found at https://github.com/google/starlark-go/blob/master/doc/spec.md.
+
+::: tip
+More examples can be found at https://github.com/tensorchord/envdlib/tree/main/src.
+:::
+
+## Using predefined package by envdlib
+
+envd provided a set of predefined libraries which are commonly used in machine learning tasks. You can find them at https://github.com/tensorchord/envdlib. 
```

Comment:
library catalog  may be discussed in another place.

ML team can build there owner libraries for there field, does it need to be mentioned here?